src/main/java/picard/fingerprint/CheckFingerprint.java

```diff
@@ -380,7 +385,15 @@ protected int doWork() {
         return super.customCommandLineValidation();
     }
 
-    static boolean isBamOrSamFile(final File f) {
-        return (BamFileIoUtils.isBamFile(f) || f.getName().endsWith(IOUtil.SAM_FILE_EXTENSION));
+    static boolean isBamOrSam(final String s) {
+        return isBamOrSam(new File(s));
+    }
+
+    static boolean isBamOrSam(final File f) {
+        return isBamOrSam(f.toPath());
+    }
+
+    static boolean isBamOrSam(final Path p) {
+        return (p.toUri().getRawPath().endsWith(BamFileIoUtils.BAM_FILE_EXTENSION) || p.toUri().getRawPath().endsWith(IOUtil.SAM_FILE_EXTENSION));
     }
```

Comment:
This is really ugly since any other tool will need to do the same. Is there a better way?

Reply:
not many tools take in vcf or sam/bam to the same argument and then do different things with it....

Reply:
got it, I'm dumb.